### PR TITLE
feat: add metadata for segment and sequence solutions

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

File: bindings/python/test/trajectory/test_segment.py [hunk 1]
```diff
@@ -174,5 +192,18 @@ def test_solve(
...
-        assert solution.condition_is_satisfied
+        assert solution.condition_is_satisfied is True
+        assert solution.segment_type is not None
+
+        assert solution.access_start_instant() is not None
+        assert solution.access_end_instant() is not None
+
+        assert solution.get_initial_mass() is not None
+        assert solution.get_final_mass() is not None
+
```
Comment (kyle-cochran) [inline bindings/python/test/trajectory/test_segment.py:207]:
```suggestion

assert solution.get_propagation_duration() is not None
```
missing or nah?

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp [hunk 2]
```diff
@@ -11,20 +14,86 @@ namespace trajectory
...
+      conditionIsSatisfied(aConditionIsSatisfied),
+      segmentType(aSegmentType)
+{
+}
+
+const Instant& Segment::Solution::accessStartInstant() const
+{
+    if (states.isEmpty())
+    {
+        throw ostk::core::error::RuntimeError("States are empty.");
```
Comment (kyle-cochran) [inline src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp:41]:
```suggestion
        throw ostk::core::error::RuntimeError("No solution available.");
```

Is this condition relevant when the Segment hasn't been solved yet?

If so, maybe an error that says something like "Segment not yet solved." would be a little more helpful.

Also, for this one specifically, is it possible for the startInstant to be available when there aren't states or nah?

**Review by kyle-cochran — APPROVED** [review]
Some nits. Sick dude!

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp (hunk 2 above)
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp:41]: Yeah good point. Technically if we run solve it should atleast have 1 state even if the condition is met immediately.

**Review by vishwa2710 — COMMENTED** [review]
Sick catches. Thanks man!

**Pull request merged**